### ISSUE TITLE
NIP199 - a simple username password login

### ISSUE THREAD

**Issue by earonesty** [nostr-protocol/nips#639]
simplified username/password login for nostr:

- create new username and password:

   - `stretched-key = pbkdf2(c=100000, dklen=32, salt=username, data=password)` 
   - post a kind `30669` event:
      - content="",
      - tags:
	 	 - "private", `aes(nostr-private-key, stretched-key)`
		 - "recover, `aes(stringify(["u": username, "p": password]), private-key)`
         - "d", `hex(sha256(stretched-key))`

- login using username
	- run a pbkdf2(c=100000, dklen=32) to get a stretched-key
    - query for kind `30669`, "d", `hex(sha256(stretched-key))`
    - decrypt "private" tag

- recover password using backup
    - query for kind `30669`, "author", `public-key`
    - decrypt "recover" tag
    
- change password:
    - (same as create!)

**Comment by fiatjaf** [comment]
If you present people with a username/password login they will assume there is a way to click on a "forget password" button.

**Comment by earonesty** [comment]
yeah I think it's important to make sure that the users know that that's not possible unless they save their backup which you should present them with as a thing to print out like BIP 39 words
